src/uct/ib/base/ib_iface.c

```diff
@@ -695,39 +767,28 @@ uct_ib_iface_roce_is_reachable(const uct_ib_device_gid_info_t *local_gid_info,
                                             &local_gid_info->gid,
                                             &sa_local) != UCS_OK)) {
         uct_iface_fill_info_str_buf(
-               params, "Couldn't convert local RoCE address to socket address");
+               params, "couldn't convert local RoCE address to socket address");
         return 0;
     }
 
     if (uct_ib_device_roce_gid_to_sockaddr(remote_ib_addr_af,
                                            remote_ib_addr + 1,
                                            &sa_remote) != UCS_OK) {
         uct_iface_fill_info_str_buf(
-               params, "Couldn't convert remote RoCE address to socket address");
+               params, "couldn't convert remote RoCE address to socket address");
         return 0;
     }
 
-    matched = ucs_sockaddr_is_same_subnet((struct sockaddr*)&sa_local,
-                                          (struct sockaddr*)&sa_remote,
-                                          prefix_bits);
-
-    if (ucs_log_is_enabled(UCS_LOG_LEVEL_DEBUG)) {
-        uct_ib_iface_log_subnet_info(&sa_local, &sa_remote, prefix_bits, matched);
+    if (iface->config.reachability_mode == UCT_IB_REACHABILITY_MODE_ROUTE) {
+        return uct_ib_iface_roce_is_routable(dev, port_num,
+                                             local_gid_info->gid_index,
+                                             (struct sockaddr *)&sa_remote,
+                                             params);
     }
 
-    if (!matched) {
-        uct_iface_fill_info_str_buf(
-                    params,
-                    "IP addresses do not match with a %u-bit prefix. local IP"
-                    " is %s, remote IP is %s",
-                    prefix_bits,
-                    ucs_sockaddr_str((struct sockaddr *)&sa_local,
-                                     local_str, 128),
-                    ucs_sockaddr_str((struct sockaddr *)&sa_remote,
-                                     remote_str, 128));
-    }
-
-    return matched;
+    return uct_ib_iface_roce_is_reachable_in_local_subnet(
```

Comment:
before this line - check or assert that config.reachability_mode is local subnet